CLI: helidon dev: Bad MAVEN_OPTS generates error message lacking detail

```
build.date               2021-04-30 13:25:54 PDT
build.version            2.2.0
build.revision           17f7cba0
project.version          1.0-SNAPSHOT
project.helidon.version  2.3.0
project.flavor           MP
latest.helidon.version   2.3.0
```

If I have a bad `MAVEN_OPTS` setting and use `helidon dev` the resulting error message is not very helpful:

```
export MAVEN_OPTS="garbage"
helidon dev starting .error: mvn io.helidon.build-tools:helidon-cli-maven-plugin:2.2.1:dev -Ddev.clean=false -Ddev.fork=false -Ddev.terminalMode=true failed with exit code 1
```

If I run the plugin manually I see what the problem is:
```
 mvn io.helidon.build-tools:helidon-cli-maven-plugin:2.2.1:dev -Ddev.clean=false -Ddev.fork=false -Ddev.terminalMode=true
Error: Could not find or load main class garbage
Caused by: java.lang.ClassNotFoundException: garbage
```

--- a/common/common/src/main/java/io/helidon/build/common/ProcessMonitor.java
+++ b/common/common/src/main/java/io/helidon/build/common/ProcessMonitor.java
@@ -29,10 +29,10 @@ import java.util.concurrent.ForkJoinPool;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Predicate;
-import java.util.stream.Collectors;
 
 import static java.util.Collections.emptyList;
 import static java.util.Objects.requireNonNullElseGet;
@@ -49,12 +49,15 @@ public final class ProcessMonitor {
     private final List<String> capturedOutput;
     private final List<String> capturedStdOut;
     private final List<String> capturedStdErr;
+    private final AtomicInteger stdOutLineCount;
+    private final AtomicInteger stdErrLineCount;
     private final Consumer<String> monitorOut;
     private final ProcessBuilder.Redirect stdIn;
     private final Consumer<String> stdOut;
     private final Consumer<String> stdErr;
     private final Predicate<String> filter;
     private final Function<String, String> transform;
+    private final String errorMsgSuffix;
     private final AtomicBoolean running;
     private volatile Process process;
     private volatile MonitorTask out;
@@ -82,6 +85,7 @@ public final class ProcessMonitor {
         private Consumer<String> stdErr;
         private Predicate<String> filter;
         private Function<String, String> transform;
+        private String errorMsgSuffix;
 
         private Builder() {
         }
@@ -188,6 +192,18 @@ public final class ProcessMonitor {
             return this;
         }
 
+
+        /**
+         * Sets error message suffix to append if there is no stdout or stderr output.
+         *
+         * @param suffix The suffix.
+         * @return This builder.
+         */
+        public Builder errorMessageSuffixIfNoOutput(String suffix) {
+            this.errorMsgSuffix = suffix;
+            return this;
+        }
+
         /**
          * Builds the instance.
          *
@@ -228,8 +244,11 @@ public final class ProcessMonitor {
         this.capturedOutput = capturing ? new ArrayList<>() : emptyList();
         this.capturedStdOut = capturing ? new ArrayList<>() : emptyList();
         this.capturedStdErr = capturing ? new ArrayList<>() : emptyList();
+        this.stdOutLineCount = new AtomicInteger();
+        this.stdErrLineCount = new AtomicInteger();
         this.filter = builder.filter;
         this.transform = builder.transform;
+        this.errorMsgSuffix = builder.errorMsgSuffix;
         this.running = new AtomicBoolean();
     }
 
@@ -269,12 +288,12 @@ public final class ProcessMonitor {
         if (stdIn != null) {
             builder.redirectInput(stdIn);
         }
-        Log.debug("Executing command: %s", builder.command().stream().collect(Collectors.joining(" ")));
+        Log.debug("Executing command: %s", String.join(" ", builder.command()));
         process = builder.start();
         Log.debug("Process ID: %d", process.pid());
         running.set(true);
-        out = monitor(process.getInputStream(), filter, transform, capturing ? this::captureStdOut : stdOut, running);
-        err = monitor(process.getErrorStream(), filter, transform, capturing ? this::captureStdErr : stdErr, running);
+        out = monitor(process.getInputStream(), filter, transform, capturing ? this::captureStdOut : this::countStdOut, running);
+        err = monitor(process.getErrorStream(), filter, transform, capturing ? this::captureStdErr : this::countStdErr, running);
         return this;
     }
 
@@ -401,6 +420,26 @@ public final class ProcessMonitor {
     }
 
     /**
+     * Returns count of any lines written to stdout.
+     *
+     * @return The count.
+     */
+    public int stdOutCount() {
+        assertStarted();
+        return stdOutLineCount.get();
+    }
+
+    /**
+     * Returns count of any lines written to stderr.
+     *
+     * @return The count.
+     */
+    public int stdErrCount() {
+        assertStarted();
+        return stdErrLineCount.get();
+    }
+
+    /**
      * Process exception.
      */
     public static class ProcessException extends Exception {
@@ -504,6 +543,9 @@ public final class ProcessMonitor {
         } else {
             message.append(" failed with exit code ").append(process.exitValue());
         }
+        if (errorMsgSuffix != null && stdOutCount() == 0 && stdErrCount() == 0) {
+            message.append(". ").append(errorMsgSuffix);
+        }
         if (capturing) {
             message.append(EOL);
             capturedOutput.forEach(line -> message.append("    ").append(line).append(EOL));
@@ -518,8 +560,18 @@ public final class ProcessMonitor {
     private static void devNull(String line) {
     }
 
-    private void captureStdOut(String line) {
+    private void countStdOut(String line) {
         stdOut.accept(line);
+        stdOutLineCount.incrementAndGet();
+    }
+
+    private void countStdErr(String line) {
+        stdErr.accept(line);
+        stdErrLineCount.incrementAndGet();
+    }
+
+    private void captureStdOut(String line) {
+        countStdOut(line);
         synchronized (capturedOutput) {
             capturedOutput.add(line);
             capturedStdOut.add(line);
@@ -527,7 +579,7 @@ public final class ProcessMonitor {
     }
 
     private void captureStdErr(String line) {
-        stdErr.accept(line);
+        countStdErr(line);
         synchronized (capturedOutput) {
             capturedOutput.add(line);
             capturedStdErr.add(line);

--- a/common/maven/src/main/java/io/helidon/build/common/maven/MavenCommand.java
+++ b/common/maven/src/main/java/io/helidon/build/common/maven/MavenCommand.java
@@ -216,7 +216,7 @@ public class MavenCommand {
             // Could not determine the Maven version. The code to do so is fragile and is known
             // not to work in some environments (especially where shims are involved). So
             // don't fail if we can't determine the maven version.
-            Log.debug("Could not determine Maven version: " + ex.toString()
+            Log.debug("Could not determine Maven version: " + ex
                     + " Assuming version is acceptable.");
             return;
         }
@@ -241,6 +241,7 @@ public class MavenCommand {
                                                       .stdErr(stdErr)
                                                       .filter(filter)
                                                       .transform(transform)
+                                                      .errorMessageSuffixIfNoOutput("Retry with --verbose or --debug option.")
                                                       .capture(false)
                                                       .build()
                                                       .start();

